src/features/settings/appearance/PostView.tsx

```diff
@@ -1,10 +1,32 @@
-import { IonLabel, IonList } from "@ionic/react";
+import type { IonActionSheetCustomEvent } from "@ionic/core";
+import {
+  ActionSheetButton,
+  IonLabel,
+  IonList,
+  IonActionSheet,
+} from "@ionic/react";
 import { ListHeader } from "./TextSize";
 import { InsetIonItem } from "../../user/Profile";
-import { useAppSelector } from "../../../store";
+import { useAppSelector, useAppDispatch } from "../../../store";
+import { useState } from "react";
 import { startCase } from "lodash";
+import { OPostAppearanceType, setPostAppearance } from "./appearanceSlice";
+import { PostAppearanceType } from "../../../services/db";
+import { OverlayEventDetail } from "@ionic/react/dist/types/components/react-component-lib/interfaces";
+
+const BUTTONS: ActionSheetButton<PostAppearanceType>[] = Object.values(
+  OPostAppearanceType
+).map(function (postAppearanceType) {
+  return {
+    text: startCase(postAppearanceType),
+    data: postAppearanceType,
+  } as ActionSheetButton<PostAppearanceType>;
+});
 
 export default function PostView() {
+  const [open, setOpen] = useState(false);
```

Comment:
This is probably going to need to be refactored out into a separate component for each option.

Or maybe we could add an abstraction of InsetIonItem+IonActionSheet for a given option. That might be even better.

Thoughts?

We can always merge this now and deal with that later.

Reply:
I agree. If we're planning to use this for the long-term it'll be worth splitting it out. I figured that since there aren't many settings currently, a one-off was enough for now. I didn't go too far since I'm fairly new to React.

Reply:
Sounds good. We can merge this for now and abstract later